src/linked_list.cxx

```diff
@@ -26,6 +26,20 @@
 //___________________________________________________________
 
 
+/*###########################################################*/
+/* listDelete() - deletes memory allocated to a linked list  */
+/* This function must be called on any list to deallocate    */
+/* the dynamic memory used in creating and manipultating the */
+/* list. (added by NJF)                                                     */
+/*###########################################################*/
+extern void listDelete(struct wetting_front* head)
+{
+  if( head != NULL ){
+    struct wetting_front *next = head->next;
+    free( head );
+    listDelete(next);
+  }
```

Comment:
Sadly, C++ does not guarantee tail call optimization, so we don't want to risk blowing up the stack
```suggestion
  while (head != NULL) {
    struct wetting_front *next = head->next;
    free( head );
    head = next;
  }
```

Reply:
If the wetting front list is large enough to blow up the stack on a recursive walk, then we probably need to consider a different data structure for the wetting fronts anyways...but either way, I added this suggestion...